src/target/cortexm.c

```diff
@@ -355,8 +566,32 @@ bool cortexm_probe(ADIv5_AP_t *ap)
 	t->attach = cortexm_attach;
 	t->detach = cortexm_detach;
 
+	/* Probe for FP extension. */
+	bool is_cortexmf = false;
+	uint32_t cpacr = target_mem_read32(t, CORTEXM_CPACR);
+	cpacr |= 0x00F00000; /* CP10 = 0b11, CP11 = 0b11 */
+	target_mem_write32(t, CORTEXM_CPACR, cpacr);
+	if (target_mem_read32(t, CORTEXM_CPACR) == cpacr)
+		is_cortexmf = true;
+
+
 	/* Should probe here to make sure it's Cortex-M3 */
-	t->tdesc = tdesc_cortex_m;
+
+	// Find the buffer size needed for the target description string we need to send to GDB,
+	// and then compute the string itself.
+	size_t size_needed;
+	char *buffer;
+	if (!is_cortexmf) {
```

Comment:
Does it make sense, as these allocate, to move building the tdesc buffers to cortexm_attach() (w/ cortexm_detach() free()'ing the buffers)? Thinking about allocation lifetimes and trying to minimise the RAM overheads here

Reply:
That might make sense. We were treating these strings as "lazily allocated" semi-static, but you're right that that multiple attaches and detaches can happen throughout the firmware program's lifetime. But in that case it might actually make the most sense to store each target description as a static global pointer, which is checked upon attach and allocated and initialized if it hasn't been already, that way each kind of tdesc is only allocated and initialized once.

Would that be acceptable or would you prefer we `malloc()` and `free()` on attach and detach respectively for each target, as you suggested?

Reply:
We'd prefer an alloc/free on every attach/detach as for processors that have multiple cores or long scan-chains, we're liable to run out of memory if users are also running flashing commands as those allocate some fairly large buffers too. Basically, concerned with getting a repeat of #960 (#1093 being the partial fix)